src/Qir/Controller/Executable/ClangClient.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Utility;
+
+namespace Microsoft.Quantum.Qir.Executable
+{
+    public class ClangClient : IClangClient
+    {
+        private const string LinkFlag = " -l ";
+        private readonly ILogger logger;
+
+        public ClangClient(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
+        public async Task CreateExecutableAsync(string[] inputFiles, string[] libraries, string libraryPath, string outputPath)
+        {
+            var inputsArg = string.Join(' ', inputFiles);
+            var librariesArg = $"{LinkFlag} {string.Join(LinkFlag, libraries)}";
```

Comment:
Is the first `{LinkFlag}` intended?

Reply:
It is - string.Join does not add one before the first element.

Reply:
I see.

Maybe there is a better alternative to using `string.Join` because adding the `{LinkFlag}` at the beginning looks like an edge case that doesn't really exists. Using Linq `ForEach` and a `StringBuilder` might be a good alternative.